bin-src/tasks/upload.js

```diff
@@ -166,6 +175,51 @@ export const uploadStorybook = async (ctx, task) => {
 
   ctx.uploadedBytes = total;
   ctx.isolatorUrl = new URL('/iframe.html', domain).toString();
+}
+
+async function uploadAsZipFile(ctx, task) {
+  const zipped = await makeZipFile(ctx);
+  const { path, size: total } = zipped;
+  const { getZipUploadUrl } = await ctx.client.runQuery(TesterGetZipUploadUrlMutation);
+  const { domain, url, sentinelUrl } = getZipUploadUrl;
+
+  task.output = starting(ctx).output;
+
+  try {
+    await uploadZip(ctx, path, url, total, (progress) => {
+      if (ctx.options.interactive) {
+        const percentage = Math.round((progress / total) * 100);
+        task.output = uploading({ percentage }).output;
+      }
+    });
+  } catch (e) {
+    if (path === e.message) {
+      throw new Error(failed({ path }).output);
+    }
+    throw e;
+  }
+
+  ctx.uploadedBytes = total;
+  ctx.isolatorUrl = new URL('/iframe.html', domain).toString();
+
+  return waitForUnpack(ctx, sentinelUrl);
+}
+
+export const uploadStorybook = async (ctx, task) => {
+  if (ctx.skip) return;
+  transitionTo(preparing)(ctx, task);
+
+  if (ctx.options.zip) {
+    try {
+      await uploadAsZipFile(ctx, task);
+    } catch (err) {
+      ctx.log.debug('error uploading zip file', err);
+      await uploadAsIndividualFiles(ctx, task);
+    }
```

Comment:
If someone passes the `--zip` flag in and it fails with the fallback to the individual uploads, is there any indication to the end user outside of running with the debug flag? Not sure if developers would care so that is probably overkill so asking out of curiosity.

Reply:
The only indication should be a resetting of the upload progress bar, unless `--debug` is passed.